.github/workflows/deploy.yaml

```diff
@@ -0,0 +1,57 @@
+name: Build and upload to PyPI
```

Comment:
pypa has some really good docs on this it was a surprising experience: https://packaging.python.org/en/latest/guides/publishing-package-distribution-releases-using-github-actions-ci-cd-workflows/

This will run a build runner every time we push/PR, and also whenever there's a release. If we push a new tag it will upload the build to testpypi for us to check before a release, if we pusblish a release it'll be automatically pushed to pypi.

What needs to be done here is:
1. Uploading the relevant github secret API tokens for testpypi and pypi.
2. Checking that the testpypi upload works (create a fake tag for an alpha version of 1.0.1 upload it and then delete the tag?)
3. Checking that the pypi upload works on publication of the next release.